add support for authenticating users

agerlic opened the pull request:
Proposition to add user authentication based on http://pusher.com/docs/auth_signatures for issue #2

Maybe AuthenticateUser method could be added to Channel instead of Client like the pusher ruby gem (https://github.com/pusher/pusher-gem/blob/master/lib/pusher/channel.rb#L139).

agerlic committed:
add support for autenticating users

jnunemaker commented:
I would rename AuthenticateUser to Authenticate and add it to channel. Since it requires channel name as the first arg, it makes more sense on channel to me. I also would like to see this library line up closely with the pusher ruby gem as I have found it to be pleasant to work with in the past.

jnunemaker commented:
We will probably have to pass an instance of the client to the channel then, so we have the key/secret and a way to make other client requests, but that seems fine.

agerlic committed:
rename AuthenticateUser to Authenticate and add it to channel
add channelAuthenticate example

agerlic commented:
I've renamed and added the Authenticate method to Channel.

I've created a new example file channelAuthenticate, which loads pusher package locally instead of git repo, I guess it could help to fix examples.
